closure/goog/crypt/sha1.js

```diff
@@ -184,19 +198,19 @@ goog.crypt.Sha1.prototype.compress_ = function(buf, opt_offset) {
       }
     }
 
-    var t = (((a << 5) | (a >>> 27)) + f + e + k + W[i]) & 0xffffffff;
+    t = ((((a << 5) | (a >>> 27)) + f + e + k + W[i & 15]) & 0xffffffff) | 0;
     e = d;
     d = c;
-    c = ((b << 30) | (b >>> 2)) & 0xffffffff;
+    c = (((b << 30) | (b >>> 2)) & 0xffffffff) | 0;
     b = a;
     a = t;
   }
 
-  this.chain_[0] = (this.chain_[0] + a) & 0xffffffff;
-  this.chain_[1] = (this.chain_[1] + b) & 0xffffffff;
-  this.chain_[2] = (this.chain_[2] + c) & 0xffffffff;
-  this.chain_[3] = (this.chain_[3] + d) & 0xffffffff;
-  this.chain_[4] = (this.chain_[4] + e) & 0xffffffff;
+  this.chain_[0] = ((this.chain_[0] + a) & 0xffffffff) | 0;
```

Comment:
As I understand the spec, the result of a bitwise & should already be a signed int32, so I'm puzzled as to how |0 will be any different (at least from my own experiments on Chrome).  Is this handling some edge case in another browser, or asm.js or something?